opus/application/apps/ui/test_ui.py

```diff
@@ -31,38 +31,47 @@ def tearDown(self):
             ###################################################
 
-    def test__api_last_blog_update_no_request(self):
-        "[test_ui.py] api_last_blog_update: no request"
+    def test__api_notifications_no_request(self):
+        "[test_ui.py] api_notifications: no request"
         with self.assertRaisesRegex(Http404,
-            r'Internal error \(No request was provided\) for /__lastblogupdate.json'):
-            api_last_blog_update(None)
+            r'Internal error \(No request was provided\) for /__notifications.json'):
+            api_notifications(None)
 
-    def test__api_last_blog_update_no_get(self):
-        "[test_ui.py] api_last_blog_update: no GET"
-        request = self.factory.get('__lastblogupdate.json')
+    def test__api_notifications_no_get(self):
+        "[test_ui.py] api_notifications: no GET"
+        request = self.factory.get('__notifications.json')
         request.GET = None
         with self.assertRaises(Http404):
-            api_last_blog_update(request)
+            api_notifications(request)
 
-    def test__api_last_blog_update_ok(self):
-        "[test_ui.py] api_last_blog_update: normal"
+    def test__api_notifications_ok(self):
+        "[test_ui.py] api_notifications: normal"
         settings.OPUS_LAST_BLOG_UPDATE_FILE = 'test_api/data/lastblogupdate.txt'
-        request = self.factory.get('__lastblogupdate.json')
-        ret = api_last_blog_update(request)
+        request = self.factory.get('__notifications.json')
+        ret = api_notifications(request)
         print(ret)
-        self.assertEqual(ret.content, b'{"lastupdate": "2019-JAN-01"}')
+        self.assertEqual(ret.content, b'{"lastupdate": "2019-JAN-01", "notification": null, "notification_mdate": null}')
 
-    def test__api_last_blog_update_bad(self):
-        "[test_ui.py] api_last_blog_update: missing"
+    def test__api_notifications_bad_update_file(self):
+        "[test_ui.py] api_notifications: missing last blog update file"
         settings.OPUS_LAST_BLOG_UPDATE_FILE = 'test_api/data/xyxyxyxyxyx.txt'
-        request = self.factory.get('__lastblogupdate.json')
-        ret = api_last_blog_update(request)
+        request = self.factory.get('__notifications.json')
+        ret = api_notifications(request)
         print(ret)
         print(ret.content)
-        self.assertEqual(ret.content, b'{"lastupdate": null}')
+        self.assertEqual(ret.content, b'{"lastupdate": null, "notification": null, "notification_mdate": null}')
+
+    def test__api_notifications_bad_notification_file(self):
+        "[test_ui.py] api_notifications: missing notifications.html file"
+        settings.OPUS_NOTIFICATIONS_FILE = 'test_api/data/xyxyxyxyxyx.html'
+        request = self.factory.get('__notifications.json')
+        ret = api_notifications(request)
+        print(ret)
+        print(ret.content)
+        self.assertEqual(ret.content, b'{"lastupdate": "2019-JAN-01", "notification": null, "notification_mdate": null}')
```

Comment:
This test fails because it doesn't override LAST_BLOG_UPDATE_FILE with the good test copy.